pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -1446,6 +1446,11 @@ type ObjectStoreSpec struct {
 	// is being used to create buckets. The default is empty.
 	// +optional
 	AllowUsersInNamespaces []string `json:"allowUsersInNamespaces,omitempty"`
+
+	// Hosting settings for the object store
+	// +optional
+	// +nullable
+	Hosting *ObjectStoreHostingSpec `json:"hosting,omitempty"`
```

Comment:
I don't think this needs to be nullable. Otherwise, I see at least one place in the code where we aren't checking to see if this is null before getting the value, which can cause null pointer panics.